bundles/org.openhab.binding.iaqualink/src/main/java/org/openhab/binding/iaqualink/internal/handler/IAqualinkHandler.java

```diff
@@ -0,0 +1,450 @@
+        } catch (NotAuthorizedException e) {
+            logger.debug("Credentials not valid");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Credentials not valid");
+        }
+    }
+
+    /**
+     * Starts/Restarts polling with an initial delay. This allows changes in the poll cycle for when commands are sent
+     * and we need to poll sooner then the next refresh cycle.
+     */
+    private synchronized void initPolling(int initalDelay) {
+        clearPolling();
+        pollFuture = scheduler.scheduleWithFixedDelay(() -> {
+            try {
+                pollController();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        }, initalDelay, refresh, TimeUnit.SECONDS);
+    }
+
+    /**
+     * Stops/clears this thing's polling future
+     */
+    private void clearPolling() {
+        ScheduledFuture<?> localFuture = pollFuture;
+        if (localFuture != null && !localFuture.isCancelled()) {
+            logger.trace("Canceling future");
+            localFuture.cancel(true);
+        }
+    }
+
+    /**
+     * Poll the controller for updates.
+     */
+    private void pollController() {
+        try {
+
+            JsonElement home = client.getHomeJson(serialNumber, sessionId);
+            JsonArray homeScreen = home.getAsJsonObject().getAsJsonArray("home_screen");
+            if (homeScreen != null) {
+                homeScreen.forEach(element -> {
+                    element.getAsJsonObject().entrySet().forEach(entry -> {
+                        updatedState(entry.getKey(), entry.getValue().getAsString());
+                        if (entry.getKey().endsWith("_heater")) {
+                            updatedState(entry.getKey() + "_status", entry.getValue().getAsString());
+                        }
+                    });
+                });
+            }
+
+            OneTouch[] oneTouches = client.getOneTouch(serialNumber, sessionId);
+            for (OneTouch ot : oneTouches) {
+                updatedState(ot.getName(), ot.getState());
+            }
+
+            Auxiliary[] auxes = client.getAux(serialNumber, sessionId);
+            for (Auxiliary aux : auxes) {
+                updatedState(aux.getName(), aux.getState());
+            }
+
+            if (firstRun) {
+                updateChannels(auxes, oneTouches);
+            }
+
+            firstRun = false;
+
+            if (getThing().getStatus() != ThingStatus.ONLINE) {
+                updateStatus(ThingStatus.ONLINE);
+            }
+
+        } catch (InterruptedException e) {
+            logger.debug("polling interupted", e);
+        } catch (IOException e) {
+            logger.debug("Exception polling", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (NotAuthorizedException e) {
+            logger.debug("Authorization Exception during polling", e);
```

Comment:
Does it need a status update?

Reply:
no, the logic calls configure again which will try and immediately authenticate (token may be invalid/ timed out).  If this fails the binding will go offline with the correct offline status message..